test/es6/toStringTag.js

```diff
@@ -5,299 +5,221 @@
+            for (var { tag, prototype } of explicitTagBuiltIns) {
+                checkBuiltInPrototype(tag, prototype);
             }
-            MyBoolean.prototype[Symbol.toStringTag] = "MyBoolean";
-
-            var m = new MyBoolean();
-            assert.areEqual("[object Boolean]", Object.prototype.toString.call(new Boolean()), "toString should have tag Boolean");
-            assert.areEqual("[object ~MyBoolean]", Object.prototype.toString.call(m), "toString should have tag ~MyBoolean");
        }
     },
     {
-       name: "sub class Date.toString",
-       body: function () {
-            class MyDate extends Date {
-                constructor() {
-                   super();
-                }
+        name: "Built-ins with no @@toStringTag property that have a default tag",
+        body: function () {
+            // Verify there is no @@toStringTag property.
+            for (var { tag, prototype } of [...internalSlotBuiltIns, argumentsInternalSlotBuiltIn]) {
+                WScript.Echo("Testing " + tag);
+                assert.isFalse(prototype.hasOwnProperty(Symbol.toStringTag), tag + " does not have a @@toStringTag property");
             }
-            MyDate.prototype[Symbol.toStringTag] = "MyDate";
-
-            var m = new MyDate();
-            assert.areEqual("[object Date]", Object.prototype.toString.call(new Date()), "toString should have tag Date");
-            assert.areEqual("[object ~MyDate]", Object.prototype.toString.call(m), "toString should have tag ~MyDate");
-       }
+        }
     },
     {
-       name: "sub class Error.toString",
-       body: function () {
-            class MyError extends Error {
-                constructor() {
-                   super();
-                }
+        name: "Default @@toStringTag built ins",
+        body: function () {
+            for (var { tag, instance } of [...primitives, ...explicitTagBuiltIns, ...explicitTagTypedArrayBuiltIns, ...internalSlotBuiltIns, argumentsInternalSlotBuiltIn]) {
+                WScript.Echo("Testing " + tag);
+                assert.areEqual("[object "+ tag + "]", Object.prototype.toString.call(instance), tag + " toString uses default tag");
             }
-            MyError.prototype[Symbol.toStringTag] = "MyError";
-
-            var m = new MyError();
-            assert.areEqual("[object Error]", Object.prototype.toString.call(new Error()), "toString should have tag Error");
-            assert.areEqual("[object ~MyError]", Object.prototype.toString.call(m), "toString should have tag ~MyError");
-       }
+        }
     },
     {
-       name: "sub class Function.toString",
-       body: function () {
-            class MyFunction extends Function {
-                constructor() {
-                   super();
-                }
+        name: "Proxy forwarding behavior",
+        body: function () {
+            for (var { tag, instance } of [...explicitTagBuiltIns]) {

```

Comment:
nit: you should be able to use just `of explicitTagBuiltIns` in this case, right?